plugins/kotlin-dataframe/src/org/jetbrains/kotlinx/dataframe/plugin/impl/api/select.kt

```diff
@@ -55,19 +63,134 @@ internal class And0 : AbstractInterpreter<ColumnsResolver>() {
 }
 
 internal class All0 : AbstractInterpreter<ColumnsResolver>() {
+    val Arguments.receiver: ColumnsResolver by arg()
+    override fun Arguments.interpret(): ColumnsResolver {
+        return columnsResolver { receiver.all() }
+    }
+}
+
+internal class All1 : AbstractInterpreter<ColumnsResolver>() {
     val Arguments.receiver by ignore()
     override fun Arguments.interpret(): ColumnsResolver {
-        return object : ColumnsResolver {
-            override fun resolve(df: PluginDataFrameSchema): List<ColumnWithPathApproximation> {
-                return df.columns().map {
-                    val path = ColumnPathApproximation(listOf(it.name))
-                    ColumnWithPathApproximation(path, it)
-                }
-            }
+        return columnsResolver { all() }
+    }
+}
+
+internal class All2 : AbstractInterpreter<ColumnsResolver>() {
+    val Arguments.receiver: SingleColumnApproximation by arg()
+    override fun Arguments.interpret(): ColumnsResolver {
+        return columnsResolver {
+            receiver.asColumnGroup().allCols()
         }
     }
 }
 
+internal class AllAfter0 : AbstractInterpreter<ColumnsResolver>() {
+    val Arguments.receiver: ColumnsResolver by arg()
+    val Arguments.column: SingleColumnApproximation by arg()
+    override fun Arguments.interpret(): ColumnsResolver {
+        return columnsResolver { receiver.allAfter(column) }
+    }
+}
+
+internal class AllAfter1 : AbstractInterpreter<ColumnsResolver>() {
+    val Arguments.receiver by ignore()
+    val Arguments.column: SingleColumnApproximation by arg()
+    override fun Arguments.interpret(): ColumnsResolver {
+        return columnsResolver { allAfter(column) }
+    }
+}
+
+internal class AllAfter2 : AbstractInterpreter<ColumnsResolver>() {
+    val Arguments.receiver by ignore()
+    val Arguments.column: SingleColumnApproximation by arg()
+    override fun Arguments.interpret(): ColumnsResolver {
+        return columnsResolver { allAfter(column) }
+    }
+}
+
+internal class AllAfter3 : AbstractInterpreter<ColumnsResolver>() {
+    val Arguments.receiver: SingleColumnApproximation by arg()
+    val Arguments.column: SingleColumnApproximation by arg()
+    override fun Arguments.interpret(): ColumnsResolver {
+        return columnsResolver {
+            receiver.asColumnGroup().allColsAfter(column)
+        }
+    }
+}
+
+internal class AllBefore0 : AbstractInterpreter<ColumnsResolver>() {
+    val Arguments.receiver: ColumnsResolver by arg()
+    val Arguments.column: SingleColumnApproximation by arg()
+    override fun Arguments.interpret(): ColumnsResolver {
+        return columnsResolver { receiver.allBefore(column) }
```

Comment:
Seems like it could be one class with the reference on function in some cases

Reply:
Would be cool! I tried to get the reference to allBefore, but it's strange.. Hard to get the reference to an extension function inside interface, plus we have ambiguity. So this doesn't compile

  public companion object {
        
        internal val AllColumnsSelectionDsl<*>.a: ColumnSelectionDsl<*>.(ColumnSelector<*, *>) -> ColumnSet<*> get() = ColumnsSelectionDsl<*>::allBefore
    }